esercizio_carcere/index.js

```diff
@@ -0,0 +1,90 @@
+
+
+
+
+document.addEventListener('DOMContentLoaded', async ()=>{
+
+    const usersResponse = await fetch('http://dms.cyberdelia.eu/api/v1/user');
+    const usersRaw = await usersResponse.json();
+    // const users = usersRaw.filter((user) => {
+    //     return user.firstname && user.lastname;
+    // });
+
+    //destructuring 
+    // const users = usersRaw.filter(({ firstname, lastname}) => {
+    //     return firstname && lastname;
+    // });
+
+    //instant return arrow function
+    let usersInPrison = usersRaw.filter(({ firstname, lastname}) => (firstname && lastname) );
+    let freeUsers = [];
+    const prison = document.getElementById('prison');
+    const freedom = document.getElementById('freedom');
+    
+
+
+    const judgeAction = (action, user) => {
+        if(action == 'Scarcera'){
+            usersInPrison = usersInPrison.filter((u) => {
+                return user.id != u.id;
+            });
+            freeUsers.push(user);
+            populate(usersInPrison, action, prison);
+            action = 'Incarcera';
+            populate(freeUsers, action, freedom);
+            return; 
+        }
+        if(action == 'Incarcera'){
```

Comment:
Puoi creare una costante con i due valori delle actions, in modo tale da ridurre la possibilità di sbagliare nello scrivere Scarcera o Incarcera. Quando dovrai usare i valori ti basterà scrivere ACTIONS.IMPRISON o ACTIONS.RELEASE
`const ACTIONS = {
  RELEASE: 'Scarcera',
  IMPRISON: 'Incarcera'
};`
```suggestion
        if(action == ACTIONS.IMPRISON){
```